### PR TITLE
Guaranteed delivery sample

### PR DESCRIPTION
**Pull request by aprooks**
Guaranteed delivery with default journal. I didn't use snapshotting as there are some discussions in progress #789 and akka/akka#17139

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Hmmm..... since the commit is from 14 days ago looks like our build server didn't pick it up. Weird :\

**Comment by aprooks** [comment]
Strange, as I was amending my old commit. Didn't expect date to be saved

**Comment by Aaronontheweb** [comment]
@aprooks git works in mysterious ways - looks good now. I'll pull this in tomorrow so @Horusiath has time for feedback (if necessary.)

File: src/examples/PersistenceExample/Program.cs [hunk 1]
```diff
@@ -16,16 +16,46 @@ static void Main(string[] args)
...
                 Console.ReadLine();
             }
         }
 
+        private static void GuaranteedDelivery(ActorSystem system)
+        {
+            Console.WriteLine("\n--- GUARANTEED DELIVERY EXAMPLE ---\n");
+            var delivery = system.ActorOf(Props.Create(()=> new DeliveryActor()),"delivery");
+
+            var deliverer = system.ActorOf(Props.Create(() => new GuaranteedDeliveryExampleActor(delivery.Path.ToString())));
```
Comment (Horusiath) [inline src/examples/PersistenceExample/Program.cs]: You don't need to stringify `ActorPath`, since it's parsed later in the code. Just push it through.

**Comment by Horusiath** [comment]
Only one little suggestion. Rest is good IMO. Great job!

**Comment by aprooks** [comment]
@Horusiath removed stringification

Also I've just noticed, that foo is sent twice: 
```
--- GUARANTEED DELIVERY EXAMPLE ---

sending: foo, with deliveryId: 1
Confirming delivery of message id: 1 and data: foo

SYSTEM: Throwing exception in Deliverer

sending: bar, with deliveryId: 2
Ignoring message id: 2 and data: bar
recovered foo
recovered confirmation 1
recovered bar
[ERROR][16.04.2015 12:47:20][Thread 0011][akka://example/user/$b] Controlled dev
astation
Cause: System.Exception: Controlled devastation
   exception log

SYSTEM: Enabling confirmations in 3 seconds

sending: foo, with deliveryId: 1
Ignoring message id: 1 and data: foo
sending: bar, with deliveryId: 2
sending: bar1, with deliveryId: 3
Ignoring message id: 2 and data: bar
Ignoring message id: 3 and data: bar1

SYSTEM: Enabled confirmations

Confirming delivery of message id: 2 and data: bar
Confirming delivery of message id: 3 and data: bar1
```

seems like it is sent, then confirmed by recovery, so only 1 extra message. Should there be any validation if we are in recovery mode or not, to wait for all confirmations being recovered?

**Comment by Horusiath** [comment]
@aprooks You always can check if persistent actor is at recovery phase using it's `IsRecovering` property. But now I see, that you've applied basically the same logic for both on command and on recovery receives and I'm not yet sure if it's valid at this point. I'll check that later.

**Comment by aprooks** [comment]
@Horusiath ok I'll check docs as well

**Comment by aprooks** [comment]
@Horusiath I've found issue: I shouldn't have persisted during recovery. Only message handling. I will update this later

**Pull request merged**